File: TypeScript/1476.ts
/**
 * 1476. Subrectangle Queries
 * https://leetcode.com/problems/subrectangle-queries/
 */

class SubrectangleQueries {
  private rectangle: number[][] = [];

  constructor(rectangle: number[][]) {
    this.rectangle = rectangle;
  }

  updateSubrectangle(
    row1: number,
    col1: number,
    row2: number,
    col2: number,
    newValue: number
  ): void {
    for (let r: number = row1; r <= row2; r++) {
      for (let c: number = col1; c <= col2; c++) {
        this.rectangle[r][c] = newValue;
      }
    }
  }

  getValue(row: number, col: number): number {
    return this.rectangle[row][col];
  }
}

File: Rust/1476.rs
/**
 * 1476. Subrectangle Queries
 * https://leetcode.com/problems/subrectangle-queries/
 */

struct SubrectangleQueries {
  rectangle: Vec<Vec<i32>>,
}

/**
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl SubrectangleQueries {
  fn new(rectangle: Vec<Vec<i32>>) -> Self {
    return Self { rectangle: rectangle };
  }

  fn update_subrectangle(&mut self, row1: i32, col1: i32, row2: i32, col2: i32, new_value: i32) {
    for r in row1..=row2 {
      for c in col1..=col2 {
        self.rectangle[r as usize][c as usize] = new_value;
      }
    }
  }

  fn get_value(&self, row: i32, col: i32) -> i32 {
    return self.rectangle[row as usize][col as usize];
  }
}
